upgrading/automated_upgrades.adoc

```diff
@@ -756,6 +756,25 @@ You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to
 ConfigMap can cause the upgrade to fail. The default is set to `false`. For more information, see the parameter 
 in xref:../install_config/aggregate_logging.adoc#aggregate-logging-ansible-variables[specify logging Ansible variables].
 
+. Back up disks.
+
+. Review the link:https://www.elastic.co/guide/en/elasticsearch/reference/5.5/breaking-changes-5.0.html[Elasticsearch breaking changes].
+The breaking changes contains important information you need to be aware of when migrating your application to Elasticsearch 5.0.
+
+. Run the following command to list all fields in the cluster that have a dot character in the name, such as *foo.bar*.
+For more context see https://bugzilla.redhat.com/show_bug.cgi?id=1666141#c5
++
+----
+oc exec -c elasticsearch -n $LOGGING_NS $pod -- es_util --query='_mapping?pretty&filter_path=**.mappings.*.properties' \
+  | jq '.[].mappings[].properties | keys' \
+  | jq .[] \
+  | egrep -e "\."
+----
++
+If the script returns any file name with a dot character, you must....
```

Comment:
@jcantrill What do the users need to do if they have files with dots in the name?

Reply:
@lukas-vlcek They need to?  delete those indices?

Reply:
@mburke5678 @jcantrill If they can delete them then this is the best approach to make sure they will not run into mapping conflicts when upgrading to ES 5.x. If, however, they need to keep this data then they need to investigate if these fields will cause mapping conflict (the script does not tell you that now).

Reply:
Do we want to recommend the ES Migration Helper plugin? Will this help with our issue?
https://github.com/elastic/elasticsearch-migration/tree/2.x

Reply:
@mburke5678 Probably not, this is ES plugin so it would have to packaged and supported by us. Moreover, this is a __site_ plugin which is an extra web UI interface and we would have to provide a way how customers could connect to this via web browser. Right now you can not access ES directly via browser (this would mean you have to add/configure proxy, setup user privs, ... probaby not worth it atm).

Reply:
@lukas-vlcek The mapping conflict we are concerned with here is dots in the name? 
If so, the solution would be to change the dot to another character, perhaps an underline? 
`file.name` -> `file_name` as one possible solution?

Reply:
@mburke5678 It is not that easy.

Changing field name means you break all saved queries, charts and dashboards relying on the original field name (plus you also need to handle conflict if there are already both fields "field.name" and "field_name"; then what are you going to rename the "field.name" to?).
And you also change structure and semantics of indexed document. It is not just technical detail.

What is going on here is that there is a change between ES 2x and ES 5x in how dots in field names are interpreted. In ES 2x it was possible to have the following documents indexed into the same index:

```javascript
{
  "field": 123  // "field" is of type number
}
```
```javascript
// Any dot in field name is treaded as any other valid character in the field name.
// It is just part of the field name.
{
  "field.name": "Bob"  // "field.name" is of type String
}
```
In ES 5x things look a bit different:

```javascript
{
  "field": 123  // "field" is of type number
}
```
```javascript
// Any dot in field name is always interpreted as nested structure.
{
  "field" : {  // "field" is of type Object
    "name": "Bob"  // "name" is of type String
  }
}
```
And this is conflicting because field "field" can not be of type number and Object at the same time. Note that the conflict happens only if there is ask to store both the documents in the same index.

What our script does is that it will tell you about the "field.name" field, and that's it. This is a signal telling us that there is possibility to run into mapping conflict when upgrading to ES 5x.

What users then need to do is go and investigate if there is also ES 2x mapping for just the "field" field with different type than Object (like in our case there is "field" of type number). In such case this index can not be upgraded to ES 5x automatically.

Reply:
@mburke5678 In any way, we shall point users to Elastic docs about [upgrading indices with dots in field names](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/dots-in-names.html#_upgrading_fields_with_dots_to_5_x). This does however does not tell users what to do if they run into conflicting mapping. The truth is that if they can not or do not want delete such indices they need to reindex the data and change the documents to get rid of conflicting data structure. This is always specific.
What we could do it provide some simple example. WDYT?

Reply:
@lukas-vlcek Does the script look only for `field.name` or any `field.*` fields?

Reply:
@mburke5678 It looks for any field name containing the dot. No matter where the dot is or what is before or after the dot.

Reply:
@lukas-vlcek Changed to:
_To help identify potential conflicts, OpenShift provides a script that examines your Elasticsearch fields to determine if any fields contain a dot in the name._